From dafa644bf2037d51c78377070f7211d99e07e8ab Mon Sep 17 00:00:00 2001
From: Keiichi Watanabe <keiichiw@chromium.org>
Date: Tue, 17 Mar 2020 19:40:28 +0900
Subject: [PATCH 20/61] CHROMIUM: drivers: media: virtio: Support
 V4L2_SEL_TGT_COMPOSE

Support V4L2_SEL_TGT_COMPOSE as a selection target in the g_selection
callback.

BUG=b:151703605
TEST=compile

Change-Id: Ibc156fc2c8eeda4ddb1c45d8bd65b03a0d160df8
Signed-off-by: Keiichi Watanabe <keiichiw@chromium.org>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/2107044
Reviewed-by: Alexandre Courbot <acourbot@chromium.org>
---
 drivers/media/virtio/virtio_video_device.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/media/virtio/virtio_video_device.c b/drivers/media/virtio/virtio_video_device.c
index 6fa013bfb472..4c80afece4b7 100644
--- a/drivers/media/virtio/virtio_video_device.c
+++ b/drivers/media/virtio/virtio_video_device.c
@@ -349,6 +349,7 @@ int virtio_video_g_selection(struct file *file, void *fh,
 	}
 
 	switch (sel->target) {
+	case V4L2_SEL_TGT_COMPOSE:
 	case V4L2_SEL_TGT_COMPOSE_BOUNDS:
 	case V4L2_SEL_TGT_COMPOSE_PADDED:
 		sel->r.width = info->frame_width;
@@ -356,7 +357,8 @@ int virtio_video_g_selection(struct file *file, void *fh,
 		break;
 	default:
 		v4l2_dbg(1, vvd->vv->debug, &vvd->vv->v4l2_dev,
-			 "unsupported/invalid selection target\n");
+			 "unsupported/invalid selection target: %d\n",
+			sel->target);
 		return -EINVAL;
 	}
 
-- 
2.25.1

